src/CommonGrids/CommonGrids.jl

```diff
@@ -0,0 +1,647 @@
+"""
+    CommonGrids
+
+CommonGrids contains convenience constructors for common
+grids. Constructors in this module accept integer
+keyword inputs, so they are dynamically created. You may
+want to use a different constructor if you're making the
+object in a performance-critical section, and if you know
+the type parameters at compile time.
+
+If no convenience constructor exists, then you may need to
+create a custom grid using our low-level compose-able API.
+
+
+# Transitioning to using CommonGrids
+
+You may have constructed a grid in the following way:
+
+```julia
+using ClimaComms
+using ClimaCore: DataLayouts, Geometry, Topologies, Quadratures, Domains, Meshes, Grids
+FT = Float64
+z_elem = 63
+z_min = FT(0)
+z_max = FT(1)
+radius = FT(6.371229e6)
+h_elem = 15
+n_quad_points = 4
+device = ClimaComms.device()
+context = ClimaComms.context(device)
+hypsography = Grids.Flat()
+global_geometry = Geometry.ShallowSphericalGlobalGeometry(radius)
+quad = Quadratures.GLL{n_quad_points}()
+h_mesh = Meshes.EquiangularCubedSphere(Domains.SphereDomain{FT}(radius), h_elem)
+h_topology = Topologies.Topology2D(context, h_mesh)
+z_boundary_names = (:bottom, :top)
+z_domain = Domains.IntervalDomain(
+    Geometry.ZPoint{FT}(z_min),
+    Geometry.ZPoint{FT}(z_max);
+    boundary_names = z_boundary_names,
+)
+z_mesh = Meshes.IntervalMesh(z_domain; nelems = z_elem)
+h_grid = Grids.SpectralElementGrid2D(h_topology, quad)
+z_topology = Topologies.IntervalTopology(context, z_mesh)
+z_grid = Grids.FiniteDifferenceGrid(z_topology)
+grid = Grids.ExtrudedFiniteDifferenceGrid(
+    h_grid,
+    z_grid,
+    hypsography,
+    global_geometry,
+)
+```
+
+You may re-write this as:
+
+```julia
+using ClimaCore.CommonGrids: ExtrudedCubedSphereGrid
+grid = ExtrudedCubedSphereGrid(;
+    z_elem = 63,
+    z_min = 0,
+    z_max = 1,
```

Comment:
Will this work? I often found that Integer values don't work with our spaces and we really need a float

Reply:
Yes, we type-cast everything inside to `FT` (which defaults to `Float64`), so this interface allows us to pass in `Int`s.